src/main/java/net/starschema/clouddb/jdbc/BQStatement.java

```diff
@@ -247,7 +247,8 @@ protected QueryResponse runSyncQuery(String querySql, boolean unlimitedBillingBy
                         this.connection.getUseLegacySql(),
                         !unlimitedBillingBytes ? this.connection.getMaxBillingBytes() : null,
                         SYNC_TIMEOUT_MILLIS, // we need this to respond fast enough to avoid any socket timeouts
-                        (long) getMaxRows()
+                        (long) getMaxRows(),
+                        this.connection.getLabels()
```

Comment:
So from what I understand here you pass on the labels wen creating your DB connection and then use the labels set in the connection and send them with each Query API.
- What if we want to pass on query related information (e.g `queryOrigin: "runner"` or let's imagine a trace ID) to the API, I imagine we are not recreating a connection every time ?

Reply:
@fzakaria ^

Reply:
This will probably need to be extended so that you can pass labels on the actual query -- very similar to maxBillable.
You can probably break it out anyways into two PRs